src/Context/Feed.tsx

```diff
@@ -0,0 +1,209 @@
+    kind: string,
+    activity: Activity<AT>,
+    data?: RT,
+    options?: ReactionAddOptions,
+  ) => ReactionAPIResponse<RT>;
+  /** Override reaction delete request */
+  doReactionDeleteRequest?: (id: string) => Promise<void | unknown>;
+  /** Override reactions filter request */
+  doReactionsFilterRequest?: (options: ReactionFilterConditions) => Promise<ReactionFilterAPIResponse<RT, CRT, AT, UT>>;
+  /** If true, feed shows the Notifier component when new activities are added */
+  notify?: boolean;
+  /** Read options for the API client (eg. limit, ranking, ...) */
+  options?: GetFeedOptions;
+  /** The user_id part of the feed */
+  userId?: string;
+};
+
+// TODO: remove UR
+export const FeedContext = React.createContext<FeedContextValue | UR>({});
+
+export const FeedProvider = <
+  UT extends DefaultUT = DefaultUT,
+  AT extends DefaultAT = DefaultAT,
+  CT extends UR = UR,
+  RT extends UR = UR,
+  CRT extends UR = UR,
+  PT extends UR = UR
+>({
+  children,
+  value,
+}: PropsWithChildren<{
+  value: FeedContextValue<UT, AT, CT, RT, CRT, PT>;
+}>) => <FeedContext.Provider value={value as FeedContextValue}>{children}</FeedContext.Provider>;
+
+export const useFeedContext = <
+  UT extends DefaultUT = DefaultUT,
+  AT extends DefaultAT = DefaultAT,
+  CT extends UR = UR,
+  RT extends UR = UR,
+  CRT extends UR = UR,
+  PT extends UR = UR
+>() => useContext(FeedContext) as FeedContextValue<UT, AT, CT, RT, CRT, PT>;
+
+export function Feed<
+  UT extends DefaultUT = DefaultUT,
+  AT extends DefaultAT = DefaultAT,
+  CT extends UR = UR,
+  RT extends UR = UR,
+  CRT extends UR = UR,
+  PT extends UR = UR
+>(props: FeedProps<UT, AT, CT, RT, CRT, PT>) {
+  const { analyticsClient, client, user, errorHandler, sharedFeedManagers } = useStreamContext<
+    UT,
+    AT,
+    CT,
+    RT,
+    CRT,
+    PT
+  >();
+  const { feedGroup, userId, children } = props;
+  const [_, setForceUpdateState] = useState(0); // eslint-disable-line @typescript-eslint/no-unused-vars
+
+  const manager = useMemo(() => {
+    if (!client) return null;
+    // const clientDifferent = this.props.client !== prevProps.client;
+    // const notifyDifferent = this.props.notify !== prevProps.notify;
+    // const feedDifferent = this.props.userId !== prevProps.userId || this.props.feedGroup !== prevProps.feedGroup;
+    // const optionsDifferent = !_isEqual(this.props.options, prevProps.options);
+    // if (clientDifferent || feedDifferent || optionsDifferent || notifyDifferent)
+    //   TODO: Implement
+    const feedId = client.feed(feedGroup, userId).id;
+    return (
+      sharedFeedManagers[feedId] ||
+      new FeedManager<UT, AT, CT, RT, CRT, PT>({ ...props, analyticsClient, client, user, errorHandler })
+    );
+  }, []);
+
+  useEffect(() => {
+    const forceUpdate = () => setForceUpdateState((prevState) => prevState + 1);
+    manager?.register(forceUpdate);
+    return () => manager?.unregister(forceUpdate);
+  }, [manager]);
```

Comment:
Probably unnecessary dependency since `manager` will never change based on previous `useMemo`.

Reply:
going to leave this dependency since prev useMemo needs some work which can lead to change feedManager